storage/badger/operation/dkg.go

```diff
@@ -74,3 +76,54 @@ func UpsertDKGStateForEpoch(epochCounter uint64, newState flow.DKGState) func(*b
 func RetrieveDKGStateForEpoch(epochCounter uint64, currentState *flow.DKGState) func(*badger.Txn) error {
 	return retrieve(makePrefix(codeDKGState, epochCounter), currentState)
 }
+
+// MigrateDKGEndStateFromV1 migrates the database that was used in protocol version v1 to the v2.
+// It reads already stored data by deprecated prefix and writes it to the new prefix with values converted to the new representation.
+// TODO(EFM, #6794): This function is introduced to implement a backward-compatible upgrade from v1 to v2.
+// Remove this once we complete the network upgrade.
+func MigrateDKGEndStateFromV1() func(txn *badger.Txn) error {
```

Comment:
Can we add some tests for this function? I think we should test the following:
- migrate each of the possible v1 end states
  - should migrate correctly and delete v1 prefix
- migrate a database where no v1 prefixes are defined

Reply:
[`e653253` (#6861)](https://github.com/onflow/flow-go/pull/6861/commits/e65325331bf333b49f592514e81ad68465850d86)